cmd/pulumi-test-language/interface.go

```diff
@@ -918,6 +919,100 @@ func (eng *languageTestServer) RunLanguageTest(
 			}
 		}
 
+		// Query the language plugin for what it thinks the project packages are, we expect to see the SDKs.
+		packages, err := languageClient.GetRequiredPackages(programInfo)
+		if err != nil {
+			return makeTestResponse(fmt.Sprintf("get required packages: %v", err)), nil
+		}
+		expectedPackages := []workspace.PackageDescriptor{}
+		for _, pkg := range programPackages {
+			if pkg.Name() == "pulumi" {
+				// Skip the pulumi package, the version for that is handled above.
+				continue
+			}
+
+			pkgDef, err := pkg.Definition()
+			if err != nil {
+				return makeTestResponse(fmt.Sprintf("get package definition: %v", err)), nil
+			}
+
+			var desc workspace.PackageDescriptor
+			if pkgDef.Parameterization == nil {
+				desc = workspace.PackageDescriptor{
+					PluginSpec: workspace.PluginSpec{
+						Name:    pkgDef.Name,
+						Version: pkgDef.Version,
+					},
+				}
+			} else {
+				desc = workspace.PackageDescriptor{
+					PluginSpec: workspace.PluginSpec{
+						Name:    pkgDef.Parameterization.BaseProvider.Name,
+						Version: &pkgDef.Parameterization.BaseProvider.Version,
+					},
+					Parameterization: &workspace.Parameterization{
+						Name:    pkgDef.Name,
+						Version: *pkgDef.Version,
+						Value:   pkgDef.Parameterization.Parameter,
+					},
+				}
+			}
+
+			expectedPackages = append(expectedPackages, desc)
+		}
+
+		versionsMatch := func(expected, actual *semver.Version) bool {
```

Comment:
Any reason not to hoist these out to the top-level? I think they can be

Reply:
🤷‍♂️ I like not filling the root namespace with helper functions if I can avoid it. They're only used in the loop below, and the logic is slightly different to the other "versionsMatch" function in this file.